src/components/search/SearchBar/SearchBar.tsx

```diff
@@ -140,10 +140,25 @@ export default function SearchBar(props: Props) {
   }
 
   function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
-    if (event.key === 'Enter' && inputValue === '') {
+    if (event.key === 'Enter') {
       event.preventDefault();
       event.stopPropagation();
-      onSelect(value);
+      if (inputValue === "") {
+        onSelect(value);
+      }
+      else if (options.length > 0) {
+        addValue(options[0]);
+        setOptions([]);
+        setInputValue("")
+        onSelect([...value, options[0]])
+      } else if (inputValue !== "") {
+        const searchQuery: SearchQuery = decodeSearchQueryLabel(inputValue)
```

Comment:
Unfortunately I think this might need to change because our goal in searching is that users can only search for complete classes or professors. With this change, if you type in something like "cs 123" and then press enter really fast, it will try to search for "cs 123" in the search results and then show an error when that class doesn't exist.

 If we can wait for the new options to load first, and then select the first one (like you do above, but with the updated suggestions), that would be great. If not, we can keep what we've typed and avoid doing anything else on enter if the options are empty, but we should make sure users can't search for incomplete class or professor names